pkg/osquery/tables/xfconf/xfconf.go

```diff
@@ -0,0 +1,164 @@
+	if len(users) < 1 {
+		return results, errors.New("kolide_xfconf requires at least one username to be specified")
+	}
+
+	for _, username := range users {
+		u, err := user.Lookup(username)
+		if err != nil {
+			return nil, fmt.Errorf("finding user by username '%s': %w", username, err)
+		}
+
+		rowData := map[string]string{"username": username}
+
+		for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("*")) {
+			userConfig, err := t.getUserConfig(u, dataQuery, rowData)
+			if err != nil {
+				return nil, fmt.Errorf("could not get xfconf settings for user %s and query %s: %w", username, dataQuery, err)
+			}
+
+			results = append(results, userConfig...)
+		}
+	}
+
+	return results, nil
+}
+
+func (t *XfconfQuerier) getUserConfig(u *user.User, dataQuery string, rowData map[string]string) ([]map[string]string, error) {
+	var results []map[string]string
+
+	// First, get user-specific settings
+	userConfigDir := getUserXfconfDir(u)
+	userConfigRows, err := t.getConfigFromDirectory(userConfigDir, dataQuery, rowData)
+	if err != nil {
+		return nil, fmt.Errorf("error getting config for user %s from directory %s: %w", u.Name, userConfigDir, err)
+	}
+	results = append(results, userConfigRows...)
+
+	// Then, get defaults
+	defaultDirs := getDefaultXfconfDirs()
+	for _, dir := range defaultDirs {
+		defaultConfigRows, err := t.getConfigFromDirectory(dir, dataQuery, rowData)
+		if err != nil {
+			return nil, fmt.Errorf("error getting config for user %s from default directory %s: %w", u.Name, dir, err)
+		}
+		results = append(results, defaultConfigRows...)
+	}
+
+	// Make sure we only include defaults if there isn't already a user-specific setting
+	return deduplicate(results), nil
+}
+
+func (t *XfconfQuerier) getConfigFromDirectory(dir string, dataQuery string, rowData map[string]string) ([]map[string]string, error) {
+	var results []map[string]string
+
+	matches, err := filepath.Glob(filepath.Join(dir, "*.xml"))
+	if err != nil {
+		return nil, fmt.Errorf("could not glob for files in directory %s: %w", dir, err)
+	}
+
+	flattenOpts := []dataflatten.FlattenOpts{
+		dataflatten.WithLogger(t.logger),
+		dataflatten.WithQuery(strings.Split(dataQuery, "/")),
+	}
+
+	for _, match := range matches {
+		flattened, err := parseXfconfXml(match, flattenOpts...)
+		if err != nil {
+			return nil, fmt.Errorf("could not read in xml file %s: %w", match, err)
+		}
+		rowData["channel"] = strings.TrimSuffix(filepath.Base(match), ".xml")
+
+		results = append(results, dataflattentable.ToMap(flattened, dataQuery, rowData)...)
+	}
+
+	return results, nil
+}
+
+// getDefaultXfconfDirs returns the path to xfconf per-channel xml files that contain
+// default settings. It checks for an override, otherwise defaults to /etc/xdg/...
+// See https://docs.xfce.org/xfce/xfce4-session/advanced#files_and_environment_variables.
+func getDefaultXfconfDirs() []string {
+	envDefaultDirsStr := os.Getenv("XDG_CONFIG_DIRS")
```

Comment:
Similar as before. Though this shouldn't be user specific, it feels like it should be a top level call